GoogleDataTransport/GDTCORLibrary/GDTCORStorage.m

```diff
@@ -329,11 +319,11 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 
 - (void)encodeWithCoder:(NSCoder *)aCoder {
   GDTCORStorage *sharedInstance = [self.class sharedInstance];
-  NSMutableOrderedSet<GDTCORStoredEvent *> *storedEvents = sharedInstance->_storedEvents;
+  NSMutableOrderedSet<GDTCOREvent *> *storedEvents = sharedInstance->_storedEvents;
```

Comment:
do we need to worry about backward compatibility now that the encoding and decoding types have changed?
say previous version we've encoded GDTCORStoreEvent type in the files and now read it as GDTCOREvent.

Reply:
That is a very good question. It might not be backwards compatible, so we should need to change the encoding key for all fields that we've altered the class of.

Reply:
I don't see this is being resolved. I guess it's a function question that would it be okay if such data is lost?
do we need test an old => new version transition to make sure the transition doesn't cause any error because of the inconsistency? Maybe it just returns null if it can't find the right format?

Reply:
I think it might be ok if data is lost here. When we transition to sqlite, this will occur again and then our migration story will be much more robust. We should definitely test the transition to make sure we understand the effects. My suggestion would be to base64 encode the bytes output of NSCoding singletons from a previous version of the SDK and attempt to read them in the newer version.

Reply:
I've written a regression test for Doudou and we ran through it together. The code currently, **with** her coding key name changes does not crash, it crashes **without** the coding key name changes, and it drops all events. Doudou is going to look into doing a class name migration to get the old data into the new format, but I've given her the greenlight to move forward with the data being lost if it can't be done.

Reply:
In order to decode the fieURL property in dataFuture of storedEvent objects, I recover the GDTCORDataFuture class and set its header in private folder for only decoding usage. Please review the newest commit.